pennylane_lightning/src/benchmarks/CMakeLists.txt

```diff
@@ -2,8 +2,6 @@ cmake_minimum_required(VERSION 3.14)
 
 project(pennylane_lightning_benchmarks)
 
-set(CMAKE_CXX_STANDARD 17)
```

Comment:
Same here as earlier?

Reply:
It seems like using `target_compile_feature` is recommended in modern CMake (see e.g. https://cliutils.gitlab.io/modern-cmake/chapters/features/cpp11.html). Still, if you think setting `CMAKE_CXX_STANDARD` is better for backward compatibility, I can recover it.

Reply:
I think the `set(CMAKE_CXX_STANDARD 17)` version is (to my brain) at least easier to see and parse at the start of a file. Plus, we can define this for the entire project up there, and easily swap in newer versions as needed.

While I like the newer option, I think I'd prefer this way for now (we can always update this later).

Reply:
I reverted some `set(CMAKE_CXX_STANDARD 20)`.